### ISSUE TITLE
Mattermost 7.0 and mmctl 7.0.0 Invalid Semantic Version

### ISSUE THREAD

**Issue by BlackHawk171** [mattermost/mmctl#528]
Hey

Just updated our on-prem mattermost to 7.0 and for some reason mmctl is giving me error 

`
Error: could not check version mismatch: Cannot parse version range 7.0.0.7.0.0.49ae60d154bb05e5aebc79de4cfa4daa.false: Invalid Semantic Version
`

**Comment by ynott** [comment]
Same as above.

The version I am using is as follows
```
/opt/mattermost/bin/mmctl version
mmctl:
Version:
GitCommit:	da16e02777ba86aceb098fdaeaff393b5a2fe7b7
GitTreeState:	"clean"
BuildDate:	2022-06-09T12:39:12Z
GoVersion:	go1.17
Compiler:	gc
Platform:	linux/amd64
```

**Comment by agnivade** [comment]
Hi all, apologies for the inconvenience. I've sent a fix here: https://github.com/mattermost/mmctl/pull/529. This will be backported.

**Comment by Podesta** [comment]
Thanks for the quickfix.  Do you guys plan to make a new build with the fix?

**Comment by summer-espresso** [comment]
Me too, I'm waiting for a build because I can't perform my current user and team management operations.

**Comment by summer-espresso** [comment]
@agnivade Thx for the 7.0.1 release of mattermost server, now it works ^^

**Comment by ynott** [comment]
I suspect that mmctl is regressing in Mattermost version 7.1.1.

mmctl included in version 7.1.1 package does not show `mmctl system version`.
`Cannot parse version range : Invalid Semantic Version` is displayed.

It goes as follows.

```
$ /opt/mattermost/bin/mmctl system version
Error: could not check version mismatch: Cannot parse version range : Invalid Semantic Version
```

Version of mmctl in package v7.1.1
```
$ /opt/mattermost/bin/mmctl version
mmctl:
Version:
GitCommit:	1413b77299b3b7912cd53f7cb41411502e8983ae
GitTreeState:	"clean"
BuildDate:	2022-06-29T08:08:10Z
GoVersion:	go1.17
Compiler:	gc
Platform:	linux/amd64
```

The version v7.0.1 released on Github can be viewed as follows.
```
stylez@mtmst:~/mmctl$ ./mmctl system version
Server version 7.1.1.7.1.1.300d22b6d0a30dd0d53e66987623cb46.false
```

```
stylez@mtmst:~/mmctl$ ./mmctl version
mmctl:
Version:	v7.0.1
GitCommit:	62266d165f180f6ab158026ff4931e1ba7138483
GitTreeState:	"clean"
BuildDate:	2022-06-17T14:46:46Z
GoVersion:	go1.17
Compiler:	gc
Platform:	linux/amd64
```

**Comment by ynott** [comment]
I suspect that mmctl is regressing in Mattermost version 7.1.1.

mmctl included in version 7.1.1 package does not show `mmctl system version`.
`Cannot parse version range : Invalid Semantic Version` is displayed.

It goes as follows.

```
$ /opt/mattermost/bin/mmctl system version
Error: could not check version mismatch: Cannot parse version range : Invalid Semantic Version
```

Version of mmctl in package v7.1.1
```
$ /opt/mattermost/bin/mmctl version
mmctl:
Version:
GitCommit:	1413b77299b3b7912cd53f7cb41411502e8983ae
GitTreeState:	"clean"
BuildDate:	2022-06-29T08:08:10Z
GoVersion:	go1.17
Compiler:	gc
Platform:	linux/amd64
```

The version v7.0.1 released on Github can be viewed as follows.
```
$ ./mmctl system version
Server version 7.1.1.7.1.1.300d22b6d0a30dd0d53e66987623cb46.false
```

```
$ ./mmctl version
mmctl:
Version:	v7.0.1
GitCommit:	62266d165f180f6ab158026ff4931e1ba7138483
GitTreeState:	"clean"
BuildDate:	2022-06-17T14:46:46Z
GoVersion:	go1.17
Compiler:	gc
Platform:	linux/amd64
```

**Comment by agnivade** [comment]
cc @pfltdv @phoinixgrr. This looks like the `Version` field isn't packaged again.

**Comment by pfltdv** [comment]
mmctl 7.1.1 version is not release yet. Looks like we need to change our release policy for mmctl.

**Comment by vaskosmihaylov** [comment]
Hello @agnivade and @pfltdv,

Sorry for writing in this closed issue, however, I do not see a point in opening a new issue so far. 

Can we summarise the issue a bit? I have MM 7.0.1 where I initially saw the problem:

`bash-4.4$ ./mattermost version | grep -w Version && ./mmctl version | grep -w Version
Version: 7.0.1
Version:	v7.0.0`

The issue at hand:

```
bash-4.4$ ./mmctl user list
Error: could not check version mismatch: incorrect server version format:
```

while I am successfully authenticated:

```
bash-4.4$ ./mmctl auth list

    | Active |   Name | Username |                       InstanceURL |
    |--------|--------|----------|-----------------------------------|
    |      * | DPX-MM | dpxadmin | https://url.com|
```

Next, I upgrade MM to 7.1.2 where according to the Mattermost docs it should have fixed the issue:

```
bash-4.4$ ./mattermost version | grep -w Version && ./mmctl version | grep -w Version
Version: 7.1.2
Version:	v7.0.1
```
however, the issue is still present:

```
bash-4.4$ ./mmctl user list
Error: could not check version mismatch: incorrect server version format:
```

According to my understanding the **mmctl 7.1.1 version** is not added to the tar archive from where we upgrade/install MM server. 

Lastly, the fix is to clone the https://github.com/mattermost/mmctl/pull/530, build the binary locally, and then use it. 

If I am missing something please let me know. Thank you!

**Comment by pfltdv** [comment]
The issue is solved for 7.2.0 release, it will have same versions with server
```
mmm:/srv/temp> wget https://releases.mattermost.com/7.2.0-rc3/mattermost-team-7.2.0-rc3-linux-amd64.tar.gz
--2022-08-12 15:44:10--  https://releases.mattermost.com/7.2.0-rc3/mattermost-team-7.2.0-rc3-linux-amd64.tar.gz
Resolving releases.mattermost.com (releases.mattermost.com)... 52.85.5.58, 52.85.5.43, 52.85.5.76, ...
Connecting to releases.mattermost.com (releases.mattermost.com)|52.85.5.58|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 325432191 (310M) [application/x-tar]
Saving to: ‘mattermost-team-7.2.0-rc3-linux-amd64.tar.gz’

mattermost-team-7.2.0-rc3-linux-amd64.tar.gz         100%[=====================================================================================================================>] 310,36M  7,22MB/s    in 47s     

2022-08-12 15:44:59 (6,55 MB/s) - ‘mattermost-team-7.2.0-rc3-linux-amd64.tar.gz’ saved [325432191/325432191]

mmm:/srv/temp> tar xzf mattermost-team-7.2.0-rc3-linux-amd64.tar.gz 
mmm:/srv/temp> ./mattermost/bin/mattermost version
Version: 7.2.0
Build Number: 7.2.0-rc3
Build Date: Wed Aug 10 02:50:01 UTC 2022
Build Hash: ce7f87155e814128b5ef25973805c476ba7523a3
Build Enterprise Ready: false
mmm:/srv/temp> ./mattermost/bin/mmctl version
mmctl:
Version:	v7.2.0
GitCommit:	9e6121b94efac5f6a0dfa74ec707a01c3e1f6ceb
GitTreeState:	"clean"
BuildDate:	2022-07-28T12:50:30Z
GoVersion:	go1.17
Compiler:	gc
Platform:	linux/amd64
mmm:/srv/temp> 
```

**Comment by isacikgoz** [comment]
> To make mmctl commands you may provide `--mismatch` to skip version check( @isacikgoz is this correct flag ?).

@pfltdv The `invalid semantic version` error had no workaround but upgrading the `mmctl` itself. We have a soft version check, it will just complain and you can suppress warnings with `--suppress-warnings`  flag. The error was causing any command to run because we always check the server version. 

We also have `--strict` which will only run commands if the `mmctl` version matches the server version.
